packages/next-international/src/app/middleware/index.ts

```diff
@@ -25,6 +25,15 @@ export function createI18nMiddleware<Locales extends readonly string[]>(
       if (strategy === 'rewrite') {
         const response = NextResponse.rewrite(mappedUrl);
         return addLocaleToResponse(response, locale);
+      } else if (strategy === 'rewriteDefault') {
+        if (locale === defaultLocale) {
+          const response = NextResponse.rewrite(mappedUrl);
+          return addLocaleToResponse(response, locale);
+        }
+        else {
+          const response = NextResponse.redirect(mappedUrl);
+          return addLocaleToResponse(response, locale);
+        }        
       } else {
```

Comment:
We can simplify this quite a bit:

```suggestion
      if (strategy === 'rewrite' || (strategy === 'rewriteDefault' && locale === defaultLocale)) {
        const response = NextResponse.rewrite(mappedUrl);
        return addLocaleToResponse(response, locale);      
      } else {
```